### PR TITLE
Functional tests - add test Create DB Backup

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add functional tests 'Create DB backup'
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | ` DOWNLOAD_PATH="/home/bbribri/Téléchargements" TEST_PATH="functional/BO/advancedParameters/database/dbBackup/01_CreateAndDeleteDbBackup" URL_FO=SHOP_URL npm run specific-test`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16848)

### CODE REVIEW

File: tests/puppeteer/campaigns/functional/BO/advancedParameters/database/dbBackup/01_CreateAndDeleteDbBackup.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  after(async () => {
+    await helper.closeBrowser(browser);
+    await files.deleteFile(`${global.BO.DOWNLOAD_PATH}/${dbBackupFilename}`);
+  });
+
+  // Login into BO
+  loginCommon.loginBO();
+
+  // Go db backup page
+  it('should go to database > sql manager  page', async function () {
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/functional/BO/advancedParameters/database/dbBackup/01_CreateAndDeleteDbBackup.js]:
```suggestion
  it('should go to database > sql manager page', async function () {
```

File: tests/puppeteer/campaigns/functional/BO/advancedParameters/database/dbBackup/01_CreateAndDeleteDbBackup.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    await expect(pageTitle).to.contains(this.pageObjects.dbBackupPage.pageTitle);
+  });
+
+  describe('Generate new backup', async () => {
+    it('should check number of db backups', async function () {
+      numberOfBackups = await this.pageObjects.dbBackupPage.getNumberOfElementInGrid();
+      await expect(numberOfBackups).to.equal(0);
+    });
+
+    it('should generate new db backup', async function () {
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/functional/BO/advancedParameters/database/dbBackup/01_CreateAndDeleteDbBackup.js:73]: Sometimes you start with a capital letter, sometimes not. Make up your mind ! :D

File: tests/puppeteer/campaigns/utils/helpers.js [hunk 3]
```diff
@@ -12,8 +12,8 @@ module.exports = {
   async closeBrowser(browser) {
     return browser.close();
   },
-  async setDownloadBehavior(browser) {
-    await browser._client.send('Page.setDownloadBehavior', {
+  async setDownloadBehavior(page) {
+    await page._client.send('Page.setDownloadBehavior', {
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/utils/helpers.js:16]: It doesn't break anything ?

File: tests/puppeteer/pages/FO/FObasePage.js [hunk 4]
```diff
@@ -22,7 +22,7 @@ module.exports = class Home extends CommonPage {
     this.languageSelectorDiv = '#_desktop_language_selector';
     this.languageSelectorExpandIcon = `${this.languageSelectorDiv} i.expand-more`;
     this.languageSelectorMenuItemLink = `${this.languageSelectorDiv} ul li a[data-iso-code='%LANG']`;
-     // footer links
+    // footer links
```
Comment (SimonGrn) [inline tests/puppeteer/pages/FO/FObasePage.js:25]: Nice catch, very important

File: tests/puppeteer/campaigns/utils/helpers.js (hunk 3 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/utils/helpers.js:16]: Nope, just a correction, i thought we could do it for browser but it was by page (in tests, we call `setDownloadBehavior(page)` anyway)

File: tests/puppeteer/campaigns/functional/BO/advancedParameters/database/dbBackup/01_CreateAndDeleteDbBackup.js (hunk 2 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/functional/BO/advancedParameters/database/dbBackup/01_CreateAndDeleteDbBackup.js:73]: all Describes starts with capital letter and it with `should` (in lower case letters). I don't think there is an error.

File: tests/puppeteer/campaigns/functional/BO/advancedParameters/database/dbBackup/01_CreateAndDeleteDbBackup.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/functional/BO/advancedParameters/database/dbBackup/01_CreateAndDeleteDbBackup.js]: nice catch thanks

File: tests/puppeteer/campaigns/functional/BO/advancedParameters/database/dbBackup/01_CreateAndDeleteDbBackup.js (hunk 2 above)
Comment (SimonGrn) [reply tests/puppeteer/campaigns/functional/BO/advancedParameters/database/dbBackup/01_CreateAndDeleteDbBackup.js:73]: Oh yeah makes sense

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**